File: RiotControl/WebService/RankedStatistics.cs
using System.Linq;

using Blighttp;

namespace RiotControl
{
	partial class WebService
	{
		string GetRankedStatistics(Summoner summoner)
		{
			const string caption = "Ranked Statistics";
			const string containerName = "rankedStatistics";
			string table = Markup.Diverse("", id: containerName);

			string inline = "var rankedStatistics =\n[\n";
			foreach (var champion in summoner.RankedStatistics)
			{
				int[] fields =
				{
					champion.Wins,
					champion.Losses,

					champion.Kills,
					champion.Deaths,
					champion.Assists,

					champion.MinionKills,

					champion.Gold,

					champion.TurretsDestroyed,

					champion.DamageDealt,
					champion.PhysicalDamageDealt,
					champion.MagicalDamageDealt,

					champion.DamageTaken,

					champion.DoubleKills,
					champion.TripleKills,
					champion.QuadraKills,
					champion.PentaKills,

					champion.TimeSpentDead,

					champion.MaximumKills,
					champion.MaximumDeaths,
				};
				var stringFields = from x in fields select x.ToString();
				inline += "new RankedStatistics(";
				inline += GetJavaScriptString(champion.ChampionName) + ", ";
				inline += string.Join(", ", stringFields);
				inline += "),\n";
			}
			inline += "];\n";
			inline += string.Format("initialiseContainer({0}, {1}, rankedStatistics);", GetJavaScriptString(caption), GetJavaScriptString(containerName));

			string inlineScript = Markup.InlineScript(inline);

			string output = table + inlineScript;

			return output;
		}
	}
}

File: Web/Script/RankedStatistics.js
function RankedStatistics(statistics)
{
    //Inherit methods
    this.calculateExtendedStatistics = BasicStatistics.prototype.calculateExtendedStatistics;
    this.baseAdd = BasicStatistics.prototype.add;

    //Call base constructor
    this.base = BasicStatistics;
    this.base(statistics);

    if(statistics !== undefined)
    {
        this.turretsDestroyed = statistics.TurretsDestroyed;

        this.damageDealt = statistics.DamageDealt;
        this.physicalDamageDealt = statistics.PhysicalDamageDealt;
        this.magicalDamageDealt = statistics.MagicalDamageDealt

        this.damageTaken = statistics.DamageTaken;

        this.doubleKills = statistics.DoubleKills;
        this.tripleKills = statistics.TripleKills;
        this.quadraKills = statistics.QuadraKills;
        this.pentaKills = statistics.PentaKills;

        this.timeSpentDead = statistics.TimeSpentDead;

        this.maximumKills = statistics.MaximumKills;
        this.maximumDeaths = statistics.MaximumDeaths;
    }
    else
    {
        this.turretsDestroyed = 0;

        this.damageDealt = 0;
        this.physicalDamageDealt = 0;
        this.magicalDamageDealt = 0

        this.damageTaken = 0;

        this.doubleKills = 0;
        this.tripleKills = 0;
        this.quadraKills = 0;
        this.pentaKills = 0;

        this.timeSpentDead = 0;

        this.maximumKills = 0;
        this.maximumDeaths = 0;
    }

    this.calculateExtendedStatistics();
}

RankedStatistics.prototype.add = function(statistics)
{
    this.baseAdd(statistics);

    this.turretsDestroyed += statistics.turretsDestroyed;

    this.damageDealt += statistics.damageDealt;
    this.physicalDamageDealt += statistics.physicalDamageDealt;
    this.magicalDamageDealt += statistics.magicalDamageDealt

    this.damageTaken += statistics.damageTaken;

    this.doubleKills += statistics.doubleKills;
    this.tripleKills += statistics.tripleKills;
    this.quadraKills += statistics.quadraKills;
    this.pentaKills += statistics.pentaKills;

    this.timeSpentDead += statistics.timeSpentDead;

    this.maximumKills = Math.max(this.maximumKills, statistics.maximumKills);
    this.maximumDeaths = Math.max(this.maximumDeaths, statistics.maximumDeaths);
}